src/main/scala/com/actian/spark_vector/datastream/DataStreamConnector.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 Actian Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.actian.spark_vector.datastream
+
+import java.net.InetSocketAddress
+import java.nio.channels.SocketChannel
+import java.nio.ByteBuffer
+
+import org.apache.spark.Logging
+
+import scala.reflect.ClassTag
+
+import com.actian.spark_vector.util.ResourceUtil.{ closeResourceOnFailure, closeResourceAfterUse }
+import com.actian.spark_vector.datastream.reader.DataStreamReader
+import com.actian.spark_vector.vector.ColumnMetadata
+import com.actian.spark_vector.srp.VectorSRPClient
+import com.actian.spark_vector.colbuffer.IntSize
+
+/**
+ * Container for the datastream connection header info.
+ * @note we keep just a part of this info, the rest of it such as column names and their logical/physical types
+ * is not stored in this container since we get it earlier from a JDBC query.
+ */
+private[datastream] case class DataStreamConnectionHeader(header: ByteBuffer) {
```

Comment:
So, I think it is risky to rely on the order Scala instantiates its values to determine how we read from the ByteBuffer. What I would do is define this as:
case class DataStreamConnectionHeader(statusCode: Int, numCols: int, vectorSize: Int, cols: Seq[ColumnInfo])

For the ColumnInfo structure you can use something similar to: https://github.com/ActianCorp/spark-vector/blob/dev-provider/provider/src/main/scala/com/actian/spark_vector/provider/Job.scala#L14

and then instantiate this DataStreamConnectionHeader with an apply(buffer: ByteBuffer) where you add the reading logic.